src/model/ProxyProject.py

```diff
@@ -1,103 +1,196 @@
+"""This module contains only one class with the same name."""
+
 from __future__ import annotations
+from typing import Callable
+from copy import copy
 
 from src.model.Project import Project
 from src.model.ProjectSnapshot import ProjectSnapshot
+from src.model.SnapshotError import SnapshotError
+from src.model.data.Alternative import Alternative
 from src.model.data.functions.FunctionalExpression import FunctionalExpression
 from src.model.data.functions.ErrorReport import ErrorReport
 from src.model.processing.Threshold import Threshold
 
 import pandas as pd
 
-class ProxyProject(Project):
-    def __init__(self):
-        self.__current_project: ProjectSnapshot = None
-        raise NotImplementedError  # TODO: IMPLEMENTIEREN
-
-    def save(self, path: str = None):
-        raise NotImplementedError  # TODO: IMPLEMENTIEREN
 
-    def undo(self) -> Project:
-        raise NotImplementedError  # TODO: IMPLEMENTIEREN
+class ProxyProject(Project):
```

Comment:
Doctoring missing for class, an explanation is probably necessary here. Like this maybe?
```suggestion
class ProxyProject(Project):
"""Time invariant interface for the management of the Projects. Necessary for the ProjectSnapshot functionality."""
```